### ISSUE TITLE
`group_events` fuzz test is flaky

### ISSUE THREAD

**Issue by mgeisler** [google/mdbook-i18n-helpers#150]
After #129, the `group_events` fuzz test seems to have become flaky. @dalance, would you be able to take a look?

An example failure is [here](https://github.com/google/mdbook-i18n-helpers/actions/runs/7542340221/job/20530931734), where the failing input can be minimized to

```
"````d\n```"
```

The diff of the failure is

```
<    "````d\n```\n````",
>    "```d\n```\n```",
```

meaning that `reconstruct_markdown(&events, None)` returned

~~~markdown
````d
```
````
~~~

whereas `reconstruct_markdown(&flattened_groups, None)` returned

~~~markdown
```d
```
```
~~~

My guess is that this is because the counting of consequetive `` ` `` is slightly off.

Indeed, looking at `flattened_groups` in the fuzz test, I see

```rust
[fuzz_targets/group_events.rs:22:5] &flattened_groups = [
    (
        1,
        Start(
            CodeBlock(
                Fenced(
                    Borrowed(
                        "d",
                    ),
                ),
            ),
        ),
    ),
    (
        2,
        Text(
            Borrowed(
                "`",
            ),
        ),
    ),
    (
        2,
        Text(
            Borrowed(
                "`",
            ),
        ),
    ),
```

with lots of lone `` ` `` characters.

A related question, is this not something which should be fixed in [pulldown-cmark-to-cmark](https://github.com/Byron/pulldown-cmark-to-cmark) instead of here? @dalance, could you create an issue in that repository and see if you can move the fix from here to there?

**Comment by dalance** [comment]
I've opened a PR to fix the existing token couting code.

I agree pulldown-cmark-to-cmark should construct an appropriate markdown without workaround.
I'll consider about it.

**Comment by mgeisler** [comment]
I let it run for a few million iterations on my desktop and it looks rock solid now!

Fixed by #153.
